test/gpu/octree/test_radius_search.cpp

```diff
@@ -36,172 +36,225 @@
-
-    pcl::gpu::Octree::ResultSqrDists result_sqr_distances_shared;
-    pcl::gpu::Octree::ResultSqrDists result_sqr_distances_individual;
-    pcl::gpu::Octree::ResultSqrDists result_sqr_distances_shared_indices;
-
-    //search GPU shared
-    octree_device.radiusSearch(queries_device, data.shared_radius, max_answers, result_device_shared, result_sqr_distances_shared);
-
-    //search GPU individual
-    octree_device.radiusSearch(queries_device,    radiuses_device, max_answers, result_device_individual, result_sqr_distances_individual);
-
-    //search GPU shared with indices
-    pcl::gpu::Octree::Indices indices;
-    indices.upload(data.indices);
-    octree_device.radiusSearch(queries_device, indices, data.shared_radius, max_answers, result_device_shared_indices, result_sqr_distances_shared_indices);
-
-    //search CPU
-    octree_device.internalDownload();
-
-    std::vector<std::vector<float>> host_sqr_distances_shared(data.tests_num);
-    std::vector<std::vector<float>> host_sqr_distances_individual(data.tests_num);
-
-    for(std::size_t i = 0; i < data.tests_num; ++i)
-    {
-        octree_device.radiusSearchHost(data.queries[i], data.shared_radius, host_search_shared[i], host_sqr_distances_shared[i], max_answers);
-        octree_device.radiusSearchHost(data.queries[i], data.radiuses[i],   host_search_individual[i], host_sqr_distances_individual[i], max_answers);
-    }
 
-    //verify results
-    const std::vector<float> radiuses (data.tests_num, data.shared_radius);
-    std::cout << "shared radius test : ";
-    verifyResults(host_search_shared, host_sqr_distances_shared, result_device_shared, result_sqr_distances_shared, radiuses, max_answers, data.tests_num, 1);
+    result_device.create(queries_device.size(), max_answers);
+  }
+
+  DataGenerator data;
+  const int max_answers = 333;
+  pcl::gpu::Octree::PointCloud cloud_device;
+  pcl::gpu::Octree octree_device;
+  pcl::gpu::Octree::Queries queries_device;
+  pcl::gpu::Octree::Radiuses radiuses_device;
+  pcl::gpu::Octree::ResultSqrDists result_sqr_distances;
+  pcl::gpu::NeighborIndices result_device;
+};
+
+TEST_F(PCL_OctreeGPUTest, shared_radius)
+{
+  octree_device.radiusSearch(queries_device,
+                             data.shared_radius,
+                             max_answers,
+                             result_device,
+                             result_sqr_distances);
+
+  // prepare output buffers on host
+  std::vector<std::vector<int>> host_search_shared(data.tests_num);
+  std::vector<std::vector<float>> host_sqr_distances_shared(data.tests_num);
+
+  // host search
+  octree_device.internalDownload();
+  for (std::size_t i = 0; i < data.tests_num; ++i)
+    octree_device.radiusSearchHost(data.queries[i],
+                                   data.shared_radius,
+                                   host_search_shared[i],
+                                   host_sqr_distances_shared[i],
+                                   max_answers);
```

Comment:
It appears to me you can further reduce code duplication if you move this section, in all three tests, inside `verifyResults`.

Reply:
I was trying to do that initially, but ran into some issues. Specifically, before performing the host operations, I need to call ` octree_device.internalDownload();`. I'm not sure why, but I was getting some errors when performing GPU search in the tests after calling `internalDownload()` in the `setUp()`, which meant that i had to put the host search after the GPU search, which is why I put it in here to get it to work. This was when I was still experimenting with fixtures, I meant to check that properly after getting the fixtures done, but forgot to do it. I'll check that this evening. I don't think `internalDownload()` (which essentially performs a bunch of `memCpy()`'s should prevent the GPU functions from working, so I might've screwed something up in the fixtures at the time.

Reply:
By the way you should rebase with master to get CI running again.